netmaster/daemon/daemon.go

```diff
@@ -271,33 +252,53 @@ func (d *MasterDaemon) runFollower() {
 	router := mux.NewRouter()
 	router.PathPrefix("/").HandlerFunc(slaveProxyHandler)
 
+	// Register netmaster service
+	d.registerService()
+
+	// just wait on stop channel
+	log.Infof("Listening in follower mode")
+	d.startListeners(router, d.stopFollowerChan)
+
+	log.Info("Exiting follower mode")
+}
+
+func (d *MasterDaemon) startListeners(router *mux.Router, stopChan chan bool) {
 	// acquire listener mutex
 	d.listenerMutex.Lock()
 	defer d.listenerMutex.Unlock()
 
-	// start server
+	// Create HTTP server and listener
 	server := &http.Server{Handler: router}
 	server.SetKeepAlivesEnabled(false)
+
 	listener, err := net.Listen("tcp", d.ListenURL)
 	if nil != err {
 		log.Fatalln(err)
 	}
-
+	log.Infof("Netmaster listening on %s", d.ListenURL)
 	listener = utils.ListenWrapper(listener)
+	defer listener.Close()
 
-	// start server
 	go server.Serve(listener)
 
-	// Register netmaster service
-	d.registerService()
+	listenURL := strings.Split(d.ListenURL, ":")
+	controlURL := strings.Split(d.ControlURL, ":")
 
-	// just wait on stop channel
-	log.Infof("Listening in follower mode")
-	<-d.stopFollowerChan
+	if strings.Compare(listenURL[1], controlURL[1]) != 0 {
+		ctrlListener, err := net.Listen("tcp", d.ControlURL)
```

Comment:
I think we have to do this for listernURL[0] != 0.0.0.0 && listenURL[0] != controlURL[0] as well.